packages/devtools_app/lib/src/screens/memory/panes/diff/controller/heap_diff.dart

```diff
@@ -17,10 +20,34 @@ class HeapDiffStore {
 
   DiffHeapClasses compare(AdaptedHeap heap1, AdaptedHeap heap2) {
     final couple = _HeapCouple(heap1, heap2);
-    return _store.putIfAbsent(couple, () => DiffHeapClasses(couple));
+    return _store.putIfAbsent(couple, () => _calculateDiffGaWrapper(couple));
   }
 }
 
+DiffHeapClasses _calculateDiffGaWrapper(_HeapCouple couple) {
+  late final DiffHeapClasses result;
+  ga.timeSync(
+    analytics_constants.memory,
+    analytics_constants.MemoryTime.calculateDiff,
+    syncOperation: () => result = DiffHeapClasses(couple),
+    screenMetricsProvider: () => _DiffAnalyticsMetrics(
+      objectsBefore: couple.older.data.objects.length,
+      objectsAfter: couple.younger.data.objects.length,
+    ),
+  );
+  return result;
+}
+
+class _DiffAnalyticsMetrics extends ScreenAnalyticsMetrics {
```

Comment:
any custom metrics added in a ScreenAnalyticsMetrics subclass will need to be added to the _gtagEvent, _gtagException, and the GtagEventDevTools class in _analytics_web.dart. These will also need to be added in the Analytics360 console.

For all these reasons, we should make sure any custom metrics we want to add really will add meaning to our analysis of the timed operation.

Reply:
It would also be preferable if we didn't have multiple ScreenAnalyticsMetrics subclasses for a single screen. For example, the only existing subclasses we currently have are PerformanceScreenMetrics and ProfilerScreenMetrics, so ideally we could also have MemoryScreenMetrics that hold all the custom metrics we need for the Memory screen.

Reply:
It looks to be cleaner to put the subclasses to the folder 'analytics', the same way as constants, so that analytics_web does not have to reference individual screen folders.

I will move other subclasses later, in separate PRs.

Reply:
is there a guidance to add this to the Analytics360 console?

Reply:
I think I may be the only one with Admin access to our console, so if you can make the changes to the analytics_web file, I will update the analytics 360 console when those land. The order matters here (you'll see other comments like "// metric5, // metric6", so once the changes are in the order should not be changed.